### PR TITLE
AdditionalSqrt

### PR DESCRIPTION
**Pull request by HansOlsson**
Follow-up to #4358
Making it a draft PR in case there is more missing.

### CODE REVIEW

**Comment by henrikt-ma** [comment]
Here are some regexp search results, including matches both in code and documentation:
- `\^[ ]*[-]?[ ]*0.5` : 58 matches
- `\^[ ]*[-]?[ ]*\([ ]*1[ ]*/[ ]*2[ ]*\)` : 18 matches

Many of these are not in documentation, and I'm wondering if there could be any cases where raising to 0.5 is more correct than using `sqrt`?

**Comment by HansOlsson** [comment]
Will recheck.
For the 2nd regexp in terms of unit-checking most don't seem to matter:
- Modelica.Media.Water.IF97_Utilities.BaseIF97.Transport.cond_dTp: `lambdaREL0 := TREL^(1/2)*sum(a[i]*TREL^(i - 1) for i in 1:4);` Where TREL has unit 1
- Modelica.Thermal.HeatTransfer.Components.Convection documentation Nu = 0.453*Re^(1/2)*Pr^(1/3);
- Modelica.Media.IdealGases.Common.MixtureGasNasa.gasMixtureViscosity `(8*(1 + M[i]/M[j]))^(1/2)`
- Modelica.Fluid.Dissipation.Utilities.SharedDocumentation.HeatTransfer.Channel.kc_evenGapLaminar Nu_3 = [2/(1+22*Pr)]^(1/6)*(Re*Pr*d_hyd/L)^(1/2) documentation-class
- Modelica.Fluid.Dissipation.HeatTransfer.StraightPipe.kc_laminar_KC (Re*IN_con.d_hyd/IN_con.L)^(1/2) - which has unit 1.

The following seem more concerning - but involve other exponents as well so should probably be a separate PR:
- Modelica.Media.IdealGases.Common.MixtureGasNasa.lowPressureThermalConductivity but there are lots of other unit-issues as well including other exponents. 
- Modelica.Media.IdealGases.Common.MixtureGasNasa.mixtureViscosityChung all kinds of exponents

**Comment by HansOlsson** [comment]
Have now gone through all of them, the remaining ones are:
- Documentation
- Part of class with other non-integer exponents as well. To me the minor benefit of having `sqrt(x)` instead of `x^0.5` is outweighed by the confusion caused by mixing `sqrt(x)` and `x^(1/3)`

File: Modelica/Fluid/Dissipation.mo [hunk 1]
```diff
@@ -4417,9 +4417,9 @@ This record is used as <strong>input record</strong> for the pressure loss funct
         DP := 0.5*IN_var.zeta_TOT*
           Modelica.Fluid.Dissipation.Utilities.Functions.General.SmoothPower(
                 m_flow,
-                (IN_con.dp_smooth/(0.5*IN_var.zeta_TOT*IN_var.rho))^0.5*IN_var.rho
+                sqrt(IN_con.dp_smooth/(0.5*IN_var.zeta_TOT*IN_var.rho))*IN_var.rho
             *IN_con.A_cross,
-                2)/(IN_var.rho*(IN_con.A_cross)^2);
+                2)/(IN_var.rho*sqrt(IN_con.A_cross));
```
Comment (maltelenz) [inline Modelica/Fluid/Dissipation.mo:4422]: Maybe I'm reading it wrong, but this seems to turn a `^2` into a `sqrt`?

**Review by maltelenz — CHANGES_REQUESTED** [review]

File: Modelica/Fluid/Dissipation.mo (hunk 1 above)
Comment (HansOlsson) [inline Modelica/Fluid/Dissipation.mo:4422]:
```suggestion
                2)/(IN_var.rho*(IN_con.A_cross)^2);
```

Comment (HansOlsson) [reply Modelica/Fluid/Dissipation.mo]: Yes, corrected now.

**Comment by casella** [comment]
@HansOlsson would you suggest some other reviewers to get this through?

**Comment by HansOlsson** [comment]
> @HansOlsson would you suggest some other reviewers to get this through?

I believe you can review it, as some are part of media and possibly @AHaumer as it relates to complex numbers.

**Comment by casella** [comment]
I have no strong opinion about this, I need to think about it.

**Review by MartinOtter — APPROVED** [review]

**Review by MartinOtter — APPROVED** [review]

**Comment by MartinOtter** [comment]
> I have no strong opinion about this, I need to think about it.

In general sqrt(x) should be preferred over x^0.5 (assuming both are directly mapped to the corresponding C-expressions):
- If x^0.5 is mapped to pow(x,0.5), then sqrt(x) is a specialized algorithm that is more efficient, might have better precision, and most CPUs have even dedicated square root instructions.
- Readability is better - it is much clearer that the computation is a square root computation

**Review by casella — APPROVED** [review]
LGTM

**Comment by Esther-Devakirubai** [comment]
@casella Can this be backported to maint/4.1.0? No milestone either.

**Comment by casella** [comment]
> @casella Can this be backported to maint/4.1.0? No milestone either.

Given @HansOlsson's [comment](https://github.com/modelica/ModelicaStandardLibrary/pull/4358#issue-2196937610), you can do that. It's not critical but it doesn't harm.

**Comment by Esther-Devakirubai** [comment]
Backporting this branch to maintenance branch by #4452

**Pull request merged**